packages/client-sdk-nodejs/src/internal/leaderboard-data-client.ts

```diff
@@ -120,43 +119,51 @@ export class LeaderboardDataClient implements ILeaderboardDataClient {
     return metadata;
   }
 
-  private convertMapToElementsList(elements: Map<number, number>): _Element[] {
+  private convertMapOrRecordToElementsList(
+    elements: Record<number, number> | Map<number, number>
+  ): _Element[] {
     const convertedElements: _Element[] = [];
-    elements.forEach((score, id) =>
-      convertedElements.push(new _Element({id: id, score: score}))
-    );
+    if (elements instanceof Map) {
+      elements.forEach((score, id) =>
+        convertedElements.push(new _Element({id: id, score: score}))
+      );
+    } else {
+      Object.entries(elements).forEach(element =>
+        convertedElements.push(
+          new _Element({id: Number(element[0]), score: element[1]})
+        )
+      );
+    }
     return convertedElements;
   }
 
-  public async leaderboardUpsert(
+  public async upsert(
```

Comment:
this class is internal, so it's fine to change the name here, but not crucial.

I don't see the public classes in this PR?